src/main/java/org/gradle/profiler/jfr/JfrProfiler.java

```diff
@@ -45,4 +45,9 @@ protected JvmArgsCalculator jvmArgsWithInstrumentation(ScenarioSettings settings
     private File getJfrFile(ScenarioSettings settings) {
         return new File(settings.getScenario().getOutputDir(), settings.getScenario().getProfileName() + PROFILE_JFR_SUFFIX);
     }
+
+    @Override
+    public void validate(ScenarioSettings settings, Consumer<String> reporter) {
+        validateMultipleIterationsWithCleanupAction(settings, reporter);
```

Comment:
Thinking about this some more: When the test doesn't re-use the daemon, then there can be cleanup actions. So `JfrProfiler.canRestartRecording()` should return true if the scenario doesn't re-use the daemon, right?

That is important for our first use scenarios, which have cleanup actions.

Reply:
Yes, correct.